cdap-master/src/main/java/co/cask/cdap/master/startup/ConfigurationCheck.java

```diff
@@ -191,8 +191,8 @@ private boolean isValidKafkaTopic(String key) {
 
   private boolean isValidPartitionKey(String key) {
     try {
-      LogPartitionType.valueOf(key);
-    } catch (IllegalArgumentException e) {
+      LogPartitionType.valueOf(key.toUpperCase());
+    } catch (IllegalArgumentException | NullPointerException e) {
```

Comment:
It happened to me on a cluster after beaming an upgrade. 

You are right, it should never be null, but if it is, it is better to handle it than to fail with "Check failed: null" :)